### ISSUE TITLE
About logo classification task

### ISSUE THREAD

**Issue by nTjing** [lindsey98/Phishpedia#20]
Hi Lin, you mentioned the logo classification task in your paper，I would like to ask if this classification task is to learn diverse feature vectors? One more question, how is this classification task implemented? Based on your paper, I think the classification model first capture the image features using ResNet and then connect the fully connected layer for classification. Is my understanding correct?
<img width="776" alt="image" src="https://github.com/lindsey98/Phishpedia/assets/113650779/d4265bf8-a924-4e36-a377-bdc292f1e4c1">

**Comment by lindsey98** [comment]
Hi,
  Yes, you are correct.
  During training, we train with normal softmax loss to make the
representation embeddings to better distinguish intra-brand logos and
inter-brand logos.
  During inference, we drop the classification head and only use the
representation vector for comparison.

nTjing ***@***.***> 于2023年10月18日周三 17:51写道：

**Comment by nTjing** [comment]
In 5.5 section of the paper, Op2: How does a Siamese model trained with one-stage transfer learning and two-stage transfer learning perform logo comparison (see our discussion in Section 3.2)?
What do one-stage transfer learning and two-stage transfer learning refer to respectively?
The classification task is first trained on the Logo2K+ dataset, after which it is migrated to the list of target brands, and finally the fully connected layer is replaced with another fully connected layer with n output neurons, corresponding to the number of target brands fine-tuned.
As  a result, one-stage transfer learning is to migrate the learning from the categorization task to the target brand list. And two-stage transfer learning is to replace the full connectivity layer on top of the one-stage migratory learning for fine-tuning. Is this the right way to understand it?

**Comment by lindsey98** [comment]
One-stage is only trained on the Logo2K+ dataset.
Two-stage is trained on Logo2K+ and then finetuned on our targetlist.
Both are trained as a classification task, like what you mentioned: "the
fully connected layer is replaced with another fully connected layer with n
output neurons, corresponding to the number of brands".

nTjing ***@***.***> 于2023年10月20日周五 16:23写道：

**Comment by Fujiaoji** [comment]
Hello Lin, for this I have a question. Since after finetuning based on Logo2k, you fine-tune again only for the last layer based on target list or all weights are changed.? Thanks

**Comment by lindsey98** [comment]
Hi Fujiao,
   All weights are changed.
